core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -323,6 +341,92 @@ func getRouterAccountConfig(routerProgramAddress string) chainwriter.PDALookups
 	}
 }
 
+func getAddressConstant(address string) chainwriter.AccountConstant {
+	return chainwriter.AccountConstant{
+		Address:    address,
+		IsSigner:   false,
+		IsWritable: false,
+	}
+}
+
+func getFeeQuoterConfig(offrampProgramAddress string) chainwriter.PDALookups {
+	return chainwriter.PDALookups{
+		Name:      "FeeQuoter",
+		PublicKey: getAddressConstant(offrampProgramAddress),
+		Seeds: []chainwriter.Seed{
+			{Static: []byte("reference_addresses")},
+		},
+		IsSigner:   false,
+		IsWritable: false,
+		// Reads the address from the reference addresses account
+		InternalField: chainwriter.InternalField{
+			TypeName: "ReferenceAddresses",
+			Location: "FeeQuoter",
+			IDL:      ccipOfframpIDL,
+		},
+	}
+}
+
+func getRouterConfig(offrampProgramAddress string) chainwriter.PDALookups {
+	return chainwriter.PDALookups{
+		Name:      "Router",
+		PublicKey: getAddressConstant(offrampProgramAddress),
+		Seeds: []chainwriter.Seed{
+			{Static: []byte("reference_addresses")},
+		},
+		IsSigner:   false,
+		IsWritable: false,
+		// Reads the address from the reference addresses account
+		InternalField: chainwriter.InternalField{
+			TypeName: "ReferenceAddresses",
+			Location: "Router",
+			IDL:      ccipOfframpIDL,
+		},
+	}
+}
+
+func getReferenceAddressesConfig(offrampProgramAddress string) chainwriter.PDALookups {
+	return chainwriter.PDALookups{
+		Name:      "ReferenceAddresses",
+		PublicKey: getAddressConstant(offrampProgramAddress),
+		Seeds: []chainwriter.Seed{
+			{Static: []byte("reference_addresses")},
+		},
+		IsSigner:   false,
+		IsWritable: false,
+	}
+}
+
+func getFeeBillingSignerConfig(offrampProgramAddress string) chainwriter.PDALookups {
+	return chainwriter.PDALookups{
+		Name:      "FeeBillingSigner",
+		PublicKey: getAddressConstant(offrampProgramAddress),
+		Seeds: []chainwriter.Seed{
+			{Static: []byte("fee_billing_signer")},
+		},
+		IsSigner:   false,
+		IsWritable: false,
+	}
+}
+
+func getCommonAddressLookupTableConfig(offrampProgramAddress string) chainwriter.DerivedLookupTable {
```

Comment:
Could you add a comment that explains how this is working? It could be helpful in the future to have it here, or something more general at the top of the file.